en/services/heartbeat.md

```diff
@@ -29,17 +30,35 @@ Components must regularly broadcast their `HEARTBEAT` and monitor for heartbeats
 
-A system may choose not to broadcast information if it does not detect another system, and it will continue to send messages to a system while it is receiving heartbeats. 
+A component may choose not to send or broadcast information on a channel (other than the `HEARTBEAT`) if it does not detect another system, and it will continue to send messages to a system while it is receiving heartbeats.
 Therefore it is important that systems:
 - broadcast a heartbeat even when not commanding the remote system.
 - do not broadcast a heartbeat when they are in a faulted state (i.e. do not publish a heartbeat from a separate thread that is unaware of the state of the rest of the component).
 
 
 ## Connecting to a GCS or MAVLink API {#gcs}
 
-The `HEARTBEAT` is also used by GCS (or Developer API) to determine if can *connect to a vehicle* in order to collect telemetry and send missions/commands.
+The `HEARTBEAT` may also used by GCS (or Developer API) to determine if it **can** connect to a vehicle in order to collect telemetry and send missions/commands.
 
 For example, *QGroundControl* will only connect to a vehicle system (i.e. not another GCS, gimbal, or onboard controller), and also checks that it has a non-zero system ID before displaying the vehicle connected message.
 QGC also uses the specific type of vehicle and other heartbeat information to control layout of the GUI.
 
 > **Note** The specific code for connecting to *QGroundControl* can be found in [MultiVehicleManager.cc](https://github.com/mavlink/qgroundcontrol/blob/master/src/Vehicle/MultiVehicleManager.cc) (see `void MultiVehicleManager::_vehicleHeartbeatInfo`).
+
+## Component Identity
+
+The _type_ ([MAV_TYPE](#MAV_TYPE)) of a component is obtained from its [HEARTBEAT.type](#HEARTBEAT) field.
+- A flight controller component will use a `MAV_TYPE` corresponding to a particular vehicle - e.g. `MAV_TYPE_FIXED_WING`, `MAV_TYPE_QUADROTOR` etc. (the use of any of these "vehicle types" indicates the component is a flight controller).
+- Any other component should use its actual type, e.g. `MAV_TYPE_GIMBAL`, `MAV_TYPE_BATTERY`, etc.
+
+Every component must have a system-unique component id, which is used for routing and for identifying multiple instances of a particular component type.
+
+> **Warning** Historically the component id was also used to determine the component type.
+  New code must not make any assumption about the type from the id used (type is determined from `HEARTBEAT.type`).
+
+MAVLink recommends that *by default* components use a type-appropriate component id from [MAV_COMPONENT](../messages/common.md#MAV_COMPONENT), and provide an interface to change the component id if needed.
+For example, a gimbal component might use use [MAV_COMP_ID_GIMBAL](../messages/common.md#MAV_COMP_ID_GIMBAL) or [MAV_COMP_ID_GIMBAL`n`](../messages/common.md#MAV_COMP_ID_GIMBAL2), and should not use `MAV_COMP_ID_GPS2`.
```

Comment:
I would use the example of camera here, and not gimbal because gimbal can be confusing since there are gimbal manager (not a component) and gimbal device (a component).